File: 프로그래머스/연습문제/Lv3/거스름돈/거스름돈.py
def solution(n, money):
    DP = [0 for i in range(n+1)]
    DP[0] = 1

    money = sorted(money)
    for m in money:
        for i in range(m, n+1):
            DP[i] += DP[i-m]

    return DP[n]

File: 프로그래머스/연습문제/Lv3/거스름돈/거스름돈.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(int n, vector<int> money) {
    vector<int> DP(n+1, 0);
    DP[0] = 1;
    
    sort(money.begin(), money.end());
    for(int m : money){
        for(int i=m; i<=n; i++)
            DP[i] += DP[i-m];
    }
    
    return DP[n];
}
